src/burnchains/mod.rs

```diff
@@ -320,6 +326,48 @@ pub struct PoxConstants {
     _shadow: PhantomData<()>,
 }
 
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub struct ExitContractConstants {
+    // The chosen exit reward cycle must have this percent of stackers that voted for it or for a lower reward cycle to be confirmed
+    pub veto_confirmation_percent: u8,
+    // This percent of mined blocks in the veto period must have vetoed the proposal for the veto to succeed.
+    pub vote_confirmation_percent: u8,
+    // For a vote to be valid in a certain reward cycle, this percent of STX must be stacked
+    pub percent_stacked_stx_for_valid_vote: u8,
+    // The lowest allowable exit reward cycle.
+    pub absolute_minimum_exit_rc: u64,
+    // Votes for reward cycles less than the current reward cycle plus this buffer are not allowed.
+    pub minimum_rc_buffer_from_present: u64,
+    // Votes for reward cycles greater than the current reward cycle plus this buffer are not allowed.
+    pub maximum_rc_buffer_from_present: u64,
+}
+
+impl ExitContractConstants {
+    /// Default constants were chosen in the exit contract proposal in the stacks-blockchain repo.
+    pub fn new(
+        veto_confirmation_percent: u8,
+        vote_confirmation_percent: u8,
+        percent_stacked_stx_for_valid_vote: u8,
+    ) -> ExitContractConstants {
+        ExitContractConstants {
+            veto_confirmation_percent,
+            vote_confirmation_percent,
+            percent_stacked_stx_for_valid_vote,
+            absolute_minimum_exit_rc: EXIT_RC_ABSOLUTE_MINIMUM_EXIT_RC,
+            minimum_rc_buffer_from_present: EXIT_RC_MINIMUM_RC_BUFFER_FROM_PRESENT,
+            maximum_rc_buffer_from_present: EXIT_RC_MAXIMUM_RC_BUFFER_FROM_PRESENT,
+        }
+    }
+
+    pub fn mainnet_default() -> ExitContractConstants {
```

Comment:
I just want to confirm this, because "veto" and "vote" have a very similar spelling.  For mainnet, the _veto_ confirmation percent is 60% -- as in, 60% of stackers must reject the upgrade?  And, the _vote_ confirmation percent is 50% -- as in, 50% of the stackers must support the upgrade?

I'd recommend using different words than "veto" and "vote" so we don't accidentally introduce a consensus bug because of a typo.  Maybe "support" and "reject" instead?

Reply:
Great point - the spelling is way too similar. I'll update it now.

Reply:
And yes, your interpretation of the numbers is correct.